src/mscorlib/shared/System/IO/Path.Unix.cs

```diff
@@ -23,6 +23,8 @@ public static string GetFullPath(string path)
             if (path.Length == 0)
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
+            PathInternal.CheckInvalidPathChars(path); 
```

Comment:
You should just do the inline check here just like Windows and take the opportunity to clean out the unused code. HasIllegalCharacters() isn't used anywhere anymore, and we don't want people to slip into using it.